### ISSUE TITLE
Why the loss computed during training is Nan？

### ISSUE THREAD

**Issue by JCU777** [mimbres/neural-audio-fp#26]
I followed the actions in the readme documentation to configure the environment（Create a virtual environment via .yml), and downloaded the Dataset-mini v1.1 to ../  . But the loss calculated when running run.py training is nan. When debugging, I found that when the data passed through the front_conv layer of the FingerPrinter model, the values of the calculated tensor were all 0 or nan. What‘s wrong and why is this happening?

**Comment by mimbres** [comment]
Sorry for not giving you a good answer. Can you provide environmental information?
```
conda env export > my_env.yml
```
Or you may try creating a virtual environment without .yml option.

**Comment by Novicei** [comment]
I had the same problem when I ran it on RTX3090 with batchsize=640

**Comment by Novicei** [comment]
**name: tensorflow
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=4.5=1_gnu
  - ca-certificates=2022.4.26=h06a4308_0
  - certifi=2021.10.8=py38h06a4308_2
  - cudatoolkit=11.0.221=h6bb024c_0
  - ld_impl_linux-64=2.35.1=h7274673_9
  - libffi=3.3=he6710b0_2
  - libgcc-ng=9.3.0=h5101ec6_17
  - libgomp=9.3.0=h5101ec6_17
  - libstdcxx-ng=9.3.0=hd4cf53a_17
  - ncurses=6.3=h7f8727e_2
  - openssl=1.1.1n=h7f8727e_0
  - pip=21.2.4=py38h06a4308_0
  - python=3.8.10=h12debd9_8
  - readline=8.1.2=h7f8727e_1
  - setuptools=61.2.0=py38h06a4308_0
  - sqlite=3.38.2=hc218d9a_0
  - tk=8.6.11=h1ccaba5_0
  - wheel=0.37.1=pyhd3eb1b0_0
  - xz=5.2.5=h7b6447c_0
  - zlib=1.2.12=h7f8727e_2
  - pip:
    - absl-py==0.15.0
    - appdirs==1.4.4
    - astunparse==1.6.3
    - audioread==2.1.9
    - cachetools==5.0.0
    - cffi==1.15.0
    - charset-normalizer==2.0.12
    - click==8.1.3
    - cycler==0.11.0
    - decorator==5.1.1
    - faiss-gpu==1.6.5
    - flatbuffers==1.12
    - fonttools==4.33.3
    - gast==0.3.3
    - google-auth==2.6.6
    - google-auth-oauthlib==0.4.6
    - google-pasta==0.2.0
    - grpcio==1.32.0
    - h5py==2.10.0
    - idna==3.3
    - importlib-metadata==4.11.3
    - joblib==1.1.0
    - kapre==0.3.5
    - keras==2.8.0
    - keras-nightly==2.5.0.dev2021032900
    - keras-preprocessing==1.1.2
    - kiwisolver==1.4.2
    - libclang==14.0.1
    - librosa==0.9.1
    - llvmlite==0.38.0
    - markdown==3.3.6
    - matplotlib==3.5.1
    - numba==0.55.1
    - numpy==1.19.5
    - oauthlib==3.2.0
    - opt-einsum==3.3.0
    - packaging==21.3
    - pillow==9.1.0
    - pooch==1.6.0
    - protobuf==3.20.1
    - pyasn1==0.4.8
    - pyasn1-modules==0.2.8
    - pycparser==2.21
    - pyparsing==3.0.8
    - python-dateutil==2.8.2
    - pyyaml==5.4.1
    - requests==2.27.1
    - requests-oauthlib==1.3.1
    - resampy==0.2.2
    - rsa==4.8
    - scikit-learn==1.0.2
    - scipy==1.8.0
    - six==1.15.0
    - soundfile==0.10.3.post1
    - tensorboard==2.8.0
    - tensorboard-data-server==0.6.1
    - tensorboard-plugin-wit==1.8.1
    - tensorflow==2.4.0
    - tensorflow-estimator==2.4.0
    - tensorflow-gpu==2.4.0
    - tensorflow-io-gcs-filesystem==0.25.0
    - termcolor==1.1.0
    - tf-estimator-nightly==2.8.0.dev2021122109
    - threadpoolctl==3.1.0
    - typing-extensions==3.7.4.3
    - urllib3==1.26.9
    - wavio==0.0.4
    - werkzeug==2.1.2
    - wrapt==1.12.1
    - zipp==3.8.0
prefix:
**

**Comment by Novicei** [comment]
`name: tensorflow
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=4.5=1_gnu
  - ca-certificates=2022.4.26=h06a4308_0
  - certifi=2021.10.8=py38h06a4308_2
  - cudatoolkit=11.0.221=h6bb024c_0
  - ld_impl_linux-64=2.35.1=h7274673_9
  - libffi=3.3=he6710b0_2
  - libgcc-ng=9.3.0=h5101ec6_17
  - libgomp=9.3.0=h5101ec6_17
  - libstdcxx-ng=9.3.0=hd4cf53a_17
  - ncurses=6.3=h7f8727e_2
  - openssl=1.1.1n=h7f8727e_0
  - pip=21.2.4=py38h06a4308_0
  - python=3.8.10=h12debd9_8
  - readline=8.1.2=h7f8727e_1
  - setuptools=61.2.0=py38h06a4308_0
  - sqlite=3.38.2=hc218d9a_0
  - tk=8.6.11=h1ccaba5_0
  - wheel=0.37.1=pyhd3eb1b0_0
  - xz=5.2.5=h7b6447c_0
  - zlib=1.2.12=h7f8727e_2
  - pip:
    - absl-py==0.15.0
    - appdirs==1.4.4
    - astunparse==1.6.3
    - audioread==2.1.9
    - cachetools==5.0.0
    - cffi==1.15.0
    - charset-normalizer==2.0.12
    - click==8.1.3
    - cycler==0.11.0
    - decorator==5.1.1
    - faiss-gpu==1.6.5
    - flatbuffers==1.12
    - fonttools==4.33.3
    - gast==0.3.3
    - google-auth==2.6.6
    - google-auth-oauthlib==0.4.6
    - google-pasta==0.2.0
    - grpcio==1.32.0
    - h5py==2.10.0
    - idna==3.3
    - importlib-metadata==4.11.3
    - joblib==1.1.0
    - kapre==0.3.5
    - keras==2.8.0
    - keras-nightly==2.5.0.dev2021032900
    - keras-preprocessing==1.1.2
    - kiwisolver==1.4.2
    - libclang==14.0.1
    - librosa==0.9.1
    - llvmlite==0.38.0
    - markdown==3.3.6
    - matplotlib==3.5.1
    - numba==0.55.1
    - numpy==1.19.5
    - oauthlib==3.2.0
    - opt-einsum==3.3.0
    - packaging==21.3
    - pillow==9.1.0
    - pooch==1.6.0
    - protobuf==3.20.1
    - pyasn1==0.4.8
    - pyasn1-modules==0.2.8
    - pycparser==2.21
    - pyparsing==3.0.8
    - python-dateutil==2.8.2
    - pyyaml==5.4.1
    - requests==2.27.1
    - requests-oauthlib==1.3.1
    - resampy==0.2.2
    - rsa==4.8
    - scikit-learn==1.0.2
    - scipy==1.8.0
    - six==1.15.0
    - soundfile==0.10.3.post1
    - tensorboard==2.8.0
    - tensorboard-data-server==0.6.1
    - tensorboard-plugin-wit==1.8.1
    - tensorflow==2.4.0
    - tensorflow-estimator==2.4.0
    - tensorflow-gpu==2.4.0
    - tensorflow-io-gcs-filesystem==0.25.0
    - termcolor==1.1.0
    - tf-estimator-nightly==2.8.0.dev2021122109
    - threadpoolctl==3.1.0
    - typing-extensions==3.7.4.3
    - urllib3==1.26.9
    - wavio==0.0.4
    - werkzeug==2.1.2
    - wrapt==1.12.1
    - zipp==3.8.0
prefix:
`

**Comment by Novicei** [comment]
But don't know why, it seems to work suddenly，My configuration is CUDA11.0, CUDNN 8.0, tensorflow2.4, I hope it can help you.

**Comment by Novicei** [comment]
> But don't know why, it seems to work suddenly，My configuration is CUDA11.0, CUDNN 8.0, tensorflow2.4, I hope it can help you.

But the current results show that it is abnormally high, the first epoch reaches 83%, I don't know why.Because I've been busy lately, I can't figure out the reason for the time being.

**Comment by mimbres** [comment]
@Novicei 
> But the current results show that it is abnormally high, the first epoch reaches 83%, I don't know why.Because I've been busy lately, I can't figure out the reason for the time being.

Is the 83% for validation or actual test? It's totally normal in validation accuracy for 1s input, because the validation set consists of a few hundred 30s songs database.

**Comment by Novicei** [comment]
Is for validation.I followed the 620_lamb file you posted for training.But I look at the accuracy rate you posted, the first epoch starts at 65%. #15

**Comment by mimbres** [comment]
@Novicei It is possible that the validation accuracy is low on the first epoch with a larger batch size. However, in the end, in the 100th (or more) epoch, bsz=640 will get better validation accuracy than bsz=120. The slow training of bsz=640 suggests that the new scheduler for learning rate and temperature will be useful. However, this topic has not been discussed further in the paper.

**Comment by Novicei** [comment]
@mimbres I don't understand what you mean by that. I mean, I used your 640 configuration file for the experiment, but the verification accuracy of the output is 83% for the first epoch 1s, 100% for the 5s and above, and the accuracy for the seventh epoch 1 second is about 93% % This should not match your results in #15, I don't know what's wrong, maybe it's my environment.

**Comment by mimbres** [comment]
@Novicei Sorry, I misunderstood your question last time. 
As you mentioned, the mini-test validation accuracy (~94%) of current repo with the 640 configuration  is higher than that I reported in #15 (~83%). Let me share a new `640_lamb_ep400` result.
<img width="764" alt="640_lamb_400_current_result" src="https://user-images.githubusercontent.com/26891722/167652300-da123a4c-c4f6-4cc7-8a4f-9f4341fcb30d.png">

It is also noticeable that the val_loss scale is different from #15.

To explain, we can get different scale of `val_loss` and `val_acc` depending on the selection of validation set we use in the [mini-search-validation()](https://github.com/mimbres/neural-audio-fp/blob/ff444ce1159f0bbbab406e9accf5e7dfdb2c0c9f/model/trainer.py#L80 ). The value of `max_n_samples` and the size of validation set has changed since performing the pre-release experiment mentioned in #15. 'max_n_samples' is currently set to 3,000 for quick validation. You can set up to 25,000 as needed.

Thanks for reporting this!
